packages/button/src/react/index.tsx

```diff
@@ -146,36 +171,34 @@ const IconContainer: React.FC<IconContainerProps> = props =>
     </div>
   ) : null
 
-interface BaseButtonProps {
+export interface ButtonProps
+  extends Omit<
+    React.DetailedHTMLProps<
+      React.ButtonHTMLAttributes<HTMLButtonElement>,
+      HTMLButtonElement
+    >,
```

Comment:
Because we often wrap Button in another component that is just passing some default props like in banner or error and then also forwardRef that component. In those use cases you have to use this more complex type. I would have done it for all components but I found it led to the original prop error inconsistently.  

Seriously I still don't see the pattern. I just know what prevents it most of the time, the new HTMLPropsFor alias. If you like try find and replacing in a few uses cases running builds and until the entire repo is converted. That's basically what I did for HTMLPropsFor when I wasn't fixing the type issues resulting from the refactor or lack of strict typing.

Reply:
This is unfortunate news. I think what it means is that we still don't understand the source of the problem or have full confidence in the solution. Things are likely much better after this PR, and we should go forward, but the issue still could come back. Or, also possible, the specific "this component is missing a required `css` prop" might still exist. Or, on that last point, did you actually test every component to ensure that didn't happen now? I did not.

Reply:
Yep lots and lots and lots and lots and lots of builds for each component. Thus my mood lately

Reply:
On that note should we create a card to ideate how we might automate testing this so we don't have to manually do this or wait for an issue to be filed again? It's happened repeatedly over the last 3-4 months.

Reply:
^ card for automated test ideas works for me.